interfaces/repo.go

```diff
@@ -42,17 +48,21 @@ type Repository struct {
 	// given a plug and a slot, are they connected?
 	plugSlots map[*Plug]map[*Slot]bool
 	backends  map[SecuritySystem]SecurityBackend
+	// attributes of plugs and slots; the attributes include attribute values from
+	// the yaml and provided at runtime via interface hooks.
+	attributes map[*Plug]map[*Slot]*InterfaceAttrs
```

Comment:
Just looking at the layout of this. I wonder if we should be following the `slotPlugs` and `plugSlots` so that we can look up information about a connection from either `(plug, slot)` and `(slot, plug)` equally. We could actually store a `connectionInfo` or whatever there so that we could host attributes and know this connection exists in the first place. We already have code managing the two attributes there and we might just re-purpose them for storing connection state.